File: largest-number/largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        
        if (nums == null || nums.length == 0) { return ""; }
        if (nums.length == 1) { return String.valueOf(nums[0]); }
        
        String[] sN = new String[nums.length];
        
        for (int i = 0; i < nums.length; ++i) {
            
            sN[i] = String.valueOf(nums[i]);
        }
        
        Arrays.sort(sN, (i, j) -> (j + i).compareTo(i + j));
        
        // if 0 is the only value in the array, return "0"
        if (sN[0].charAt(0) == '0') { return "0"; }
        
        StringBuilder sB = new StringBuilder();
        
        for (String s : sN) {
            
            sB.append(s);
        }
        
        return sB.toString();
    }
}

File: largest-number/largest-number.cpp
class Solution {
public:
    string largestNumber(const vector<int>& nums) {
        
        if (nums.size() == 0) { return ""; }
        if (nums.size() == 1) { return to_string(nums[0]); }
        
        // convert nums to string vector
        vector<string> sN;
        for (int i : nums) { sN.push_back(to_string(i)); }
        
        // sort and arrange substring values to get the highest value combinations
        sort (sN.begin(), sN.end(), [](auto i, auto j){ return i+j > j+i; });
        
        // if nums has 0 as its only value, return "0"
        if (sN[0] == "0") { return "0"; }
        
        // build and return the largest number value
        string s = "";
        for (string k : sN) { s += k; }
        return s;
    }
};